### PR TITLE
Sct 182 update tier after upgrade

### PR DESCRIPTION
**Pull request by JaycobDuffel**

### CODE REVIEW

File: src/app/AccountSideBar.js [hunk 1]
```diff
@@ -1,91 +1,93 @@
...
+            userId: userId,
+            tier: data2.data.tier
+          })
+          setTierData({
+            "tier": data2.data.tier,
+            "userCount": data2.data.slack_user_list.split(" ").length,
+            "channelCount": data2.data.slack_channel_list.split(" ").length
+          })
+        }));
+
```
Comment (katiemcgoff) [inline src/app/AccountSideBar.js:56]: What changed in this code block?

**Comment by katiemcgoff** [comment]
There seems to be a lot of white-space changes checked in. This makes it difficult to review your actual changes. If you can roll back and check in only your changes, that would be great. If that is not possible, please describe your changes in the PR description and add comments in the PR at line changes that we should be reviewing specifically.

File: src/app/AccountSideBar.js [hunk 2]
```diff
@@ -1,91 +1,93 @@
...
+        })
+      ])
+        .then(axios.spread((data1, data2) => {
+          setUser({
+            name: data1.data.first_name,
+            org: data1.data.organization_data.name,
+            email: data1.data.email,
+            userId: userId,
+            teamId: teamId,
+            tier: data2.data.tier
```
Comment (JaycobDuffel) [inline src/app/AccountSideBar.js:40]: added tier to user object

File: src/app/AccountSideBar.js [hunk 3]
```diff
@@ -1,91 +1,93 @@
...
+          })
+          ls.set("user", {
+            name: data1.data.first_name,
+            org: data1.data.organization_data.name,
+            email: data1.data.email,
+            teamId: teamId,
+            userId: userId,
+            tier: data2.data.tier
```
Comment (JaycobDuffel) [inline src/app/AccountSideBar.js:48]: save tier to local storage

File: src/app/App.js [hunk 4]
```diff
@@ -8,17 +8,20 @@ import SettingsPanel from './shared/SettingsPanel';
...
 import { UserProvider } from "./contexts/UserContext"
+import { StripeProvider } from './contexts/StripeContext';
 
 const App = (props) => {
   const [state, setState] = useState({})
   const [user, setUser] = useState()
-  const providerValue = useMemo(() => ({user, setUser}), [user,setUser])
+  const [stripeSuccess, setStripeSuccess] = useState()
+  const providerValue = useMemo(() => ({ user, setUser }), [user, setUser])
+  const stripeValues = useMemo(() => ({ stripeSuccess, setStripeSuccess }), [stripeSuccess, setStripeSuccess])
```
Comment (JaycobDuffel) [inline src/app/App.js:19]: set stripeSuccess context and stripeValues

File: src/app/App.js [hunk 5]
```diff
@@ -50,13 +53,14 @@ const App = (props) => {
...
+  let navbarComponent = !state.isFullPageLayout ? <AppNavbar /> : '';
+  let sidebarComponent = !state.isFullPageLayout ? <AccountSideBar /> : '';
+  let SettingsPanelComponent = !state.isFullPageLayout ? <SettingsPanel /> : '';
+  let footerComponent = !state.isFullPageLayout ? <Footer /> : '';
 
-    return (
-      <UserProvider value={providerValue}>
+  return (
+    <UserProvider value={providerValue}>
+      <StripeProvider value={stripeValues}>
```
Comment (JaycobDuffel) [inline src/app/App.js:63]: added stripeProvider wrapper

File: src/app/Components/SquareLoader.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from 'react'
```
Comment (JaycobDuffel) [inline src/app/Components/SquareLoader.js:1]: made the square loader from example code into its own component

File: src/app/Components/Stripe/CheckoutForm.js [hunk 7]
```diff
@@ -49,75 +69,32 @@ const CheckoutForm = ({ price, setShow, user, subscriptionType }) => {
...
 
-            </div>
+                </div>
+            }
         </form>
     );
 };
 
-export default CheckoutForm
-
```
Comment (JaycobDuffel) [inline src/app/Components/Stripe/CheckoutForm.js:79]: removed unused code

File: src/app/Components/Stripe/CheckoutForm.js [hunk 8]
```diff
@@ -49,75 +69,32 @@ const CheckoutForm = ({ price, setShow, user, subscriptionType }) => {
...
-            <div>
-                <br></br>
-                <h3>Total: ${price}</h3>
-                <br></br>
-                <div style={{ display: "flex", flexDirection: "row", alignItems: "center", justifyContent: "space-between" }}>
-                    <div>
-                        <h3>{user.email}</h3>
-                    </div>
-                    <div>
+            {loading ? <SquareLoader /> :
```
Comment (JaycobDuffel) [inline src/app/Components/Stripe/CheckoutForm.js:72]: conditionally render loader when waiting for response from stripe call

File: src/app/Components/Stripe/CheckoutForm.js [hunk 9]
```diff
@@ -31,15 +40,26 @@ const CheckoutForm = ({ price, setShow, user, subscriptionType }) => {
...
         } else {
-            setShow(false);
+            // setShow(false);
             ApiService.saveStripeInfo({
                 price: price * 100,
                 email, payment_method_id: paymentMethod.id,
                 subscriptionType
             })
                 .then(response => {
+                    if (response.data.message === "Success") {
```
Comment (JaycobDuffel) [inline src/app/Components/Stripe/CheckoutForm.js:50]: if response is successful, update tier and set stripe success to true

File: src/app/Components/Stripe/CheckoutForm.js [hunk 10]
```diff
@@ -31,15 +40,26 @@ const CheckoutForm = ({ price, setShow, user, subscriptionType }) => {
...
+                        setShow(false);
+
+                        ApiService.updateOrgTier(user.teamId, data)
+                        setUser({ ...user, tier: data.tier })
+                        setLoading(false);
+                        setStripeSuccess(true);
+                    }
                     console.log(response.data);
                 }).catch(error => {
+                    setLoading(false);
```
Comment (JaycobDuffel) [inline src/app/Components/Stripe/CheckoutForm.js:62]: setLoading to false if error

**Review by DawsonDease — APPROVED** [review]

**Review by katiemcgoff — CHANGES_REQUESTED** [review]
Looks great! 

One change now that I've seen it in action: 
- display /upgrade page inside the sidebar/navbar framework

Functionality needed:
- upon successful upgrade, the user's session data needs to be updated (see screenshot).
image

**Review by katiemcgoff — APPROVED** [review]

**Pull request merged**